fhir-persistence-jdbc/src/main/java/org/linuxforhealth/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1910,6 +1910,9 @@ private ExtractedSearchParameters extractSearchParameters(Resource fhirResource,
                                 // Alternatively, we could pull the search parameter from the FHIRRegistry so we can use versioned references.
                                 // However, that would bypass search parameter filtering and so we favor the SeachUtil method here instead.
                                 SearchParameter compSP = searchHelper.getSearchParameter(p.getResourceType(), component.getDefinition());
+                                if(compSP==null){
+                                    throw new NullPointerException(String.format("The 'compSP' parameter was specified to null [ResourceType=%s,Uri=%s]",p.getResourceType(),component.getDefinition()));
+                                }                                
```

Comment:
style suggestion and unrelated comment fix
```suggestion
                                // However, that would bypass search parameter filtering and so we favor the SeachHelper method here instead.
                                SearchParameter compSP = searchHelper.getSearchParameter(p.getResourceType(), component.getDefinition());
                                if (compSP == null) {
                                    throw new NullPointerException(String.format("The 'compSP' parameter was " + 
                                        specified to null [ResourceType=%s,Uri=%s]", p.getResourceType(), component.getDefinition()));
                                }                                
```